src/main/java/com/mongodb/starter/repositories/MongoDBPersonRepository.java

```diff
@@ -0,0 +1,140 @@
+import static java.util.Arrays.asList;
+
+@Repository
+public class MongoDBPersonRepository implements PersonRepository {
+
+    private static final TransactionOptions txnOptions = TransactionOptions.builder()
+                                                                           .readPreference(ReadPreference.primary())
+                                                                           .readConcern(ReadConcern.MAJORITY)
+                                                                           .writeConcern(WriteConcern.MAJORITY)
+                                                                           .build();
+    @Autowired
+    private MongoClient client;
+    private MongoCollection<Person> personCollection;
+
+    @PostConstruct
+    void init() {
+        personCollection = client.getDatabase("test").getCollection("persons", Person.class);
+    }
+
+    @Override
+    public Person save(Person person) {
+        person.setId(new ObjectId());
+        personCollection.insertOne(person);
+        return person;
+    }
+
+    @Override
+    public List<Person> saveAll(List<Person> persons) {
+        try (ClientSession clientSession = client.startSession()) {
+            return clientSession.withTransaction(() -> {
+                persons.forEach(p -> p.setId(new ObjectId()));
+                personCollection.insertMany(clientSession, persons);
+                return persons;
+            }, txnOptions);
+        }
+    }
+
+    @Override
+    public List<Person> findAll() {
+        return personCollection.find().into(new ArrayList<>());
+    }
+
+    @Override
+    public List<Person> findAll(List<String> ids) {
+        return personCollection.find(in("_id", mapToObjectIds(ids))).into(new ArrayList<>());
+    }
+
+    @Override
+    public Person findOne(String id) {
+        return personCollection.find(eq("_id", new ObjectId(id))).first();
+    }
+
+    @Override
+    public long count() {
+        return personCollection.countDocuments();
+    }
+
+    @Override
+    public long delete(String id) {
+        return personCollection.deleteOne(eq("_id", new ObjectId(id))).getDeletedCount();
+    }
+
+    @Override
+    public long delete(List<String> ids) {
+        try (ClientSession clientSession = client.startSession()) {
+            return clientSession.withTransaction(
+                    () -> personCollection.deleteMany(clientSession, in("_id", mapToObjectIds(ids))).getDeletedCount(),
+                    txnOptions);
+        }
+    }
+
+    @Override
+    public long deleteAll() {
+        try (ClientSession clientSession = client.startSession()) {
+            return clientSession.withTransaction(
+                    () -> personCollection.deleteMany(clientSession, new BsonDocument()).getDeletedCount(), txnOptions);
+        }
+    }
+
+    @Override
+    public Person update(Person person) {
+        FindOneAndReplaceOptions options = new FindOneAndReplaceOptions();
```

Comment:
Like this:

    FindOneAndReplaceOptions options = new FindOneAndReplaceOptions()
             .returnDocument(AFTER);

Reply:
A static constructor would be even better :-) ! Ok perfect, I didn't get it the first time.

Reply:
Resolved.